File: Capstone/src/main/java/QRCODE/Capstone/domain/Member.java
package QRCODE.Capstone.domain;


import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter @Setter
public class Member {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    private Long id;

    private String username;
    private String password;

    @OneToMany(mappedBy = "member")
    private List<OldPlace> oldPlaces = new ArrayList<>();


    private String currentPlace;
    private LocalDateTime localDateTime;

    private String name;
    private int age;
    private String phone;

}

File: src/main/kotlin/org/example/redisexample/domain/entity/Member.kt
package org.example.redisexample.domain.entity

import jakarta.persistence.*
import java.io.Serializable
import java.time.LocalDateTime

@Entity
@Table(name = "members")
class Member(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    val id: Int = 0,

    @Column(name = "name", nullable = false)
    val name: String,

    @Column(name = "score", nullable = false)
    var score: Int = 0,

    @Column(name = "created_dt", nullable = false)
    val createdDt: LocalDateTime = LocalDateTime.now(),

    @Column(name = "updated_dt")
    private var updatedDt: LocalDateTime? = null
) : Serializable {
    fun updatedScore(inputScore: Int) {
        score += inputScore
        updatedDt = LocalDateTime.now()
    }
}